samples/net/mqtt_publisher/src/main.c

```diff
@@ -242,12 +245,28 @@ static void broker_init(void)
 	broker6->sin6_family = AF_INET6;
 	broker6->sin6_port = htons(SERVER_PORT);
 	inet_pton(AF_INET6, SERVER_ADDR, &broker6->sin6_addr);
+
+#if IS_ENABLED(CONFIG_MQTT_LIB_SOCKS)
```

Comment:
So here is would be:

if (IS_ENABLED(CONFIG_MQTT_LIB_SOCKS)) {
...
}

Reply:
But that won't work that easily. It uses things that are available only if the config is actually defined. So it either have to stay like this or we would have to include lots of stuff when it is not needed. That includes some defines from the configuration header, but in the later cases also fields of a struct that don't exist unless this config is enabled.

Besides that, a few lines above a similar check is made (CONFIG_NET_IPV6) and it is a compile time check too. Would the approach you're suggesting not make it too inconsistent?